websites/P/Planet Minecraft/presence.ts

```diff
@@ -0,0 +1,341 @@
+const presence = new Presence({
+		clientId: "1193411747898458132",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	stateless = "stateless";
+
+async function getStrings() {
+	return presence.getStrings({
+		viewHome: "general.viewHome",
+		viewMember: "general.viewMember",
+		viewChannel: "general.viewChannel",
+		readingThread: "general.readingThread",
+		browsing: "general.browsing",
+		reading: "general.reading",
+	});
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		details: "Viewing page:",
+		largeImageKey: "https://i.imgur.com/pNsonk0.png",
+		startTimestamp: browsingTimestamp,
+	};
+
+	if (!strings) strings = await getStrings();
+
+	const pathname = document.location.pathname.replaceAll(/\/$/g, "");
+
+	if (pathname === "") {
+		presenceData.state = strings.viewHome;
+		delete presenceData.details;
+	} else {
+		const url = `${document.location.protocol}//${document.location.host}${document.location.pathname}`;
```

Comment:
Isnt href the exact same?

Reply:
`href` includes query params which could contain sensitive data.